pkg/component/extensions/operatingsystemconfig/operatingsystemconfig.go

```diff
@@ -1048,8 +1049,12 @@ func KeyV2(
 		if kubeReserved := kubeletConfiguration.KubeReserved; kubeReserved != nil {
 			data = append(data, fmt.Sprintf("%s-%s-%s-%s", kubeReserved.CPU, kubeReserved.Memory, kubeReserved.PID, kubeReserved.EphemeralStorage))
 		}
-		if eviction := kubeletConfiguration.EvictionHard; eviction != nil && eviction.MemoryAvailable != nil {
-			data = append(data, *eviction.MemoryAvailable)
+		if eviction := kubeletConfiguration.EvictionHard; eviction != nil {
+			parts := make([]string, 0, 5)
+			for _, value := range []*string{eviction.ImageFSAvailable, eviction.ImageFSInodesFree, eviction.MemoryAvailable, eviction.NodeFSAvailable, eviction.NodeFSInodesFree} {
+				parts = append(parts, ptr.Deref(value, ""))
+			}
+			data = append(data, strings.Join(parts, "-"))
```

Comment:
Nit: I would have preferred a single `Sprintf(...)` instead of this boiler plate. Especially, the magic numbering in the sizing of the `parts` array might become useless over time if the amount of attributes changes.

Reply:
I've converted it into a sprintf statement. At least in the way the `eviction` fields are currently concatenated, adding a new field would result in a different hash. So the array size actually can't change.